src/devices/gridstream.c

```diff
@@ -0,0 +1,258 @@
+- E - Source Address
+- M - Uptime (time since last outage in seconds)
+- I - Counter
+- C - Clock
+- K - Unknown
+- X - CRC (poly 0x1021, init set by provider)
+
+*/
+
+#include "decoder.h"
+#include <stdbool.h>
+#include <stdlib.h>
+#include <time.h>
+
+static uint16_t known_crc_init[] = {
+        0xe623,
+        0x5fd6,
+        0xD553,
+        0x45F8,
+        0x62C1,
+        0x23D1,
+        0x2C22,
+        0x142A,
+};
+
+static int gridstream_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    uint8_t const preamble[] = {
+            0xAA,
+            0xAA,
+            0x00,
+    };
+    uint8_t b[BITBUF_COLS];
+    uint16_t stream_len;
+    uint16_t crc;
+    uint16_t crc_count = 0;
+    bool crc_ok        = false;
+    char destwanaddress_str[13];
+    char srcwanaddress_str[13];
+    char srcaddress_str[9];
+    char destaddress_str[9];
+    uint32_t uptime;
+    time_t clock;
+    char clock_str[80];
+    int subtype;
+    unsigned offset = bitbuffer_search(bitbuffer, 0, 0, preamble, 24) + 16;
+    /* TODO: Special handling for V4 and V5 syncwords as V5 is not compatible with extract_bytes_uart() */
+    int decoded_len = extract_bytes_uart(bitbuffer->bb[0], offset, bitbuffer->bits_per_row[0] - offset, b);
+
+    if (decoded_len >= 5) {
+        switch (b[2]) {
+        case 0x2A:
+            subtype = b[3];
+            switch (subtype) {
+            case 0x55:
+                stream_len = (b[4] << 8) | b[5];
+                if ((decoded_len - 6) < stream_len) {
+                    return DECODE_ABORT_LENGTH;
+                }
+                crc = (b[4 + stream_len] << 8) | b[5 + stream_len];
+                do {
+                    if (crc16(&b[6], stream_len - 2, 0x1021, known_crc_init[crc_count]) == crc) {
+                        crc_ok = true;
+                    }
+                    else {
+                        crc_count++;
+                    }
+                } while (crc_count < (sizeof(known_crc_init) / 2) && crc_ok == false);
+                if (!crc_ok) {
+                    decoder_log(decoder, 1, __func__, "Bad CRC or unknown init value. ");
+                    if (stream_len == 0x23) {
+                        decoder_log_bitrow(decoder, 1, __func__, &b[6], decoded_len * 8, "Use RevEng to find init value.");
+                    }
+                    return DECODE_FAIL_MIC;
+                }
+
+                sprintf(destwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[7], b[8], b[9], b[10], b[11], b[12]);
+                sprintf(srcwanaddress_str, "%02x%02x%02x%02x%02x%02x", b[13], b[14], b[15], b[16], b[17], b[18]);
+                sprintf(srcaddress_str, "%02x%02x%02x%02x", b[26], b[27], b[28], b[29]);
+                uptime = ((uint32_t)b[20] << 24) | (b[21] << 16) | (b[22] << 8) | b[23];
+
+                /* clang-format off */
+                        data = data_make(
+                                "model",       "", DATA_STRING,  "LandisGyr GridStream",
```

Comment:
The `model` should be shorter and the only non-alphanum is one dash in the middle. `LandisGyr-GS` maybe?

Reply:
Addressed in [7edd93a](https://github.com/merbanan/rtl_433/pull/2616/commits/7edd93a5aead3c669f4bd8e5dbf2d6d32e9ede96)